rllib/evaluation/postprocessing.py

```diff
@@ -217,18 +270,18 @@ def compute_gae_for_sample_batch(
         else:
             last_r = policy._value(**input_dict)
 
-    # Adds the policy logits, VF preds, and advantages to the batch,
-    # using GAE ("generalized advantage estimation") or not.
-    batch = compute_advantages(
-        sample_batch,
-        last_r,
-        policy.config["gamma"],
-        policy.config["lambda"],
-        use_gae=policy.config["use_gae"],
-        use_critic=policy.config.get("use_critic", True),
+    # Set the SampleBatch.VALUES_BOOTSTRAPPED field to VF_PREDS[1:] + the
```

Comment:
why is it not VF_PREDS[:-1]

Reply:
Then we would drop the last ts again.
You want to have in the "bootstrapped sequence" exactly the values of t+1 (for each t). Such that bootstrapped[-1] is then the actually bootstrapped value estimate.

Reply:
oh wait this is a part of the impala math, my bad.